File: tests/fixtures/sample.go
package main

// MyStruct is a sample struct.
// It has multiple fields.
type MyStruct struct {
	Name string
	age  int
}

// NewMyStruct creates a new instance.
func NewMyStruct(name string, age int) *MyStruct {
	return &MyStruct{Name: name, age: age}
}

// GetName returns the name.
func (s *MyStruct) GetName() string {
	return s.Name
}

// setAge sets the age (unexported).
func (s MyStruct) setAge(age int) {
	s.age = age
}

// Stringer is an interface.
type Stringer interface {
	String() string
}

// standalone function
func greet(name string) string {
	return "Hello, " + name
}

// multiReturn demonstrates multiple returns.
func multiReturn(x int) (int, error) {
	return x * 2, nil
}

// MaxSize is the maximum size.
const MaxSize = 1024

const (
	StatusActive   = "active"
	StatusInactive = "inactive"
)

// DefaultName is the default name.
var DefaultName = "world"

var (
	counter int
	enabled bool
)

// StringList is a type alias.
type StringList = []string

// UserID is a named type.
type UserID int64

File: tests/fixtures/sample.rs
/// A sample struct
#[derive(Debug, Clone)]
pub struct MyStruct {
    pub name: String,
    age: u32,
}

/// Implementation for MyStruct
impl MyStruct {
    /// Creates a new instance
    pub fn new(name: String, age: u32) -> Self {
        Self { name, age }
    }

    /// Gets the name
    pub fn name(&self) -> &str {
        &self.name
    }

    pub async fn fetch(&self) -> Result<Data, Error> {
        todo!()
    }

    pub unsafe fn raw_ptr(&self) -> *const u8 {
        std::ptr::null()
    }
}

/// A trait definition
pub trait Drawable {
    fn draw(&self);
    fn bounds(&self) -> Rect;
}

/// Implement Drawable for MyStruct
impl Drawable for MyStruct {
    fn draw(&self) {
        println!("drawing");
    }

    fn bounds(&self) -> Rect {
        Rect::default()
    }
}

/// A standalone function
pub fn greet(name: &str) -> String {
    format!("Hello, {}", name)
}

/// A generic function
pub fn process<T: Display + Debug>(item: T) -> Result<(), Error> {
    Ok(())
}

/// An enum
#[derive(Debug)]
pub enum Color {
    Red,
    Green,
    Blue,
    Custom(u8, u8, u8),
}

/// A constant
pub const MAX_SIZE: usize = 1024;

/// A static
static COUNTER: AtomicUsize = AtomicUsize::new(0);

/// A type alias
pub type Result<T> = std::result::Result<T, MyError>;

pub(crate) fn crate_visible() {}
fn private_fn() {}
